src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -383,7 +384,26 @@ void RaftReplService::load_repl_dev(sisl::byte_view const& buf, void* meta_cooki
     }
 
     if (rd_sb->destroy_pending == 0x1) {
-        LOGINFOMOD(replication, "ReplDev group_id={} was destroyed, skipping the load", group_id);
+        LOGINFOMOD(replication, "ReplDev group_id={} was destroyed, reclaim the stale resource", group_id);
+        // if we do not add the repl_dev to m_rd_map, it will not be permanently destroyed since gc thread finds the
+        // pending destroy repl_dev only from m_rd_map. so, we should try to reclaim all the repl_dev stale resources
```

Comment:
can we call `permanent_destroy` instead of duplicating here?

Reply:
permanent_destroy is a member function a class RaftReplDev, in which the following two line is class specified

```
    logstore_service().destroy_log_dev(m_data_journal->logdev_id());
    m_stage.update([](auto* stage) { *stage = repl_dev_stage_t::PERMANENT_DESTROYED; });
```

here, we do not have a repl dev instance, so we can not call permanent_destroy directly

Reply:
move this block down to L412 after rdev constructed?  There is no much cost in RaftReplDev constructor.

Reply:
there are some corner cases. for example, 
```
void RaftReplDev::permanent_destroy() {
    RD_LOGI("Permanent destroy for raft repl dev group_id={}", group_id_str());
    m_raft_config_sb.destroy();
    m_data_journal->remove_store();
    logstore_service().destroy_log_dev(m_data_journal->logdev_id());
    m_stage.update([](auto* stage) { *stage = repl_dev_stage_t::PERMANENT_DESTROYED; });
    // we should destroy repl_dev superblk only after all the resources are cleaned up, so that is crash recovery
    // occurs, we have a chance to find the stale repl_dev and reclaim all the stale resources.
    m_rd_sb.destroy();
}
```

if we crash after `m_data_journal->remove_store()` but before  `logstore_service().destroy_log_dev(m_data_journal->logdev_id());`, then log store is removed and ReplLogStore can not be created, [this line](https://github.com/eBay/HomeStore/blob/fba9034a2e212948970dedf617bb8b86ddf116de/src/lib/replication/repl_dev/raft_repl_dev.cpp#L40-L41) is expected to fail. as a result, the repl_dev can not be created since the log store of it is lost